Fixed OAuth and logins

XanderVeldeman opened the pull request:
# Fix for #76 
- I also added the ability to sign in using a ~Bill Gates' dick~ Microsoft account
- OAuth login now looks like this
![image](https://user-images.githubusercontent.com/25788097/57141598-dc5df300-6dba-11e9-80d9-9e306b5b0f18.png)
- OAuth uses the correct providers to collect the name and the email
- OAuth accounts are set as verified

- You may have to recreate your User account or edit the DB row

![owyeah](https://media.giphy.com/media/xUOwGnmRSbVAnDtE9a/source.gif)

XanderVeldeman committed:
Should be fixed

- Also added Microsoft login

XanderVeldeman committed:
JS entries

XanderVeldeman committed:
Finally fixed it

- username also get's filled in with a claim

XanderVeldeman committed:
Removed incorrect comment

NielsSensei reviewed (dismissed):
BUG 1:
Als ik via google wil registreren en ik pak een account kom ik op volgend scherm. In orde, als ik op de register button druk doet hij dit en maakt hij de user totaal niet aan.
![image](https://user-images.githubusercontent.com/44059248/57149287-a9702b00-6dcb-11e9-8a56-dd43efc9ab7b.png)
![image](https://user-images.githubusercontent.com/44059248/57149318-bbea6480-6dcb-11e9-834a-d912b8c9359f.png)

BUG 2 (Not sure if ya can do something bout this):
Als ik via microsoft wil registreren en ik verander van gedacht en ik druk hier op annuleren gebeurt er dit. Als ik niet op annuleren terug kom ik uiteindelijk terug op de eerste bug.
![image](https://user-images.githubusercontent.com/44059248/57149404-f2c07a80-6dcb-11e9-9872-e1fc4901ec99.png)
![image](https://user-images.githubusercontent.com/44059248/57149436-023fc380-6dcc-11e9-8b3d-9c4e4e5f7333.png)

Gewoon registreren werkt zoals ook: also KreyGasm op dit
![image](https://user-images.githubusercontent.com/44059248/57149729-acb7e680-6dcc-11e9-9279-52a66bc5f788.png)

Helaas zijn de twee bugs te belangrijk om deze te doen passen. Code Knight out
![deny](https://media.tenor.com/images/b10411f7f3a9c5df3ce39a9678eac1dd/tenor.gif)

NielsSensei commented:
EDIT: alle functies binnen Account/Manage werken nog buiten 1 which is external login via google.

PS: Authenticator app is fancy af.

XanderVeldeman committed:
Stupid mistakes were stupid

- changed field from disabled to readonly
- the controller now uses the email info provided by the provider

XanderVeldeman commented:
should be working now, moet ik een 2FA app toevoegen voor sprint 2?

NielsSensei commented:
Ik krijg soms nog deze error PR blijft open tot ik ge iets gevonden hebt. (Na selecteren acc voor login via microhard of google). Also ou externe authenticator dinge van identity werkt perfect. Ik heb het getest me me maar haar google authenticator app.

![image](https://user-images.githubusercontent.com/44059248/57165873-a17ab000-6df8-11e9-8980-d4a9a4601214.png)

NielsSensei reviewed (changes requested):
^

XanderVeldeman commented:
de authenticator is nogsteeds standaard identity, daar is nog **niks** aan aangepast

XanderVeldeman commented:
En da ziet er eerder een error uit gethrowed door de provider, zou het kunnen da ge u naam ni mee geeft bij google of microsoft authentication?
Also ge hebt toch u user en userlogins table gereset?
